[RM Ch3] Update SFC Section 3.5.5 call flows

kumarsumangit opened the pull request:
Fixes #2276 Update Service function chaining section in 3.5.1 : 3.5.1.3 call flows
On top of PR 2319 to update architecture
Adding call flows related with SFC 
Updated the figures indices in chapter 3

walterkozlowski commented:
@kumarsumangit: is this stable contents that we can do review?  It has been stale a bit for a while, and there are some conflicts.  Please remove the conflicts (probably related to png files), remove/update the contents,  and advise whether the team need to start reviewing.

kumarsumangit commented:
@walterkozlowski sorry for late work .. i have updated with changed required to resolve conflicts. Please do review the latest.

walterkozlowski reviewed (changes requested):
@kumarsumangit, the contents of this PR is valuable.2 and we will need it in Kali.  However, I need your help to remove some conflicts first, so we can review it further and eventually approve and merge. Please consider the following:
1. Your addition number from 3.5.3 to 3.5.3.3 (with subsections).  This is wrong numbering as this should be added at the end of 3.5.5 (not 3.5.3).  So could you please renumber these section to 3.5.5 (or 3.5.5.x, 3.5.5.x.y when appropriate)?  This will remove the first conflct.
2. Please number temporarily your new (very interesting) diagrams to 3.11a, 3.11b, etc.  I will renumber them later on (together with other diagrams to be renumbered) using my Admin rights.  This will help us to remove another conflict area.
3. Please remove any changes you propose to the following sections (not related to SFC), like Storage.  This will remove another source of conflicts.  Please remove the necessary adjustments to me.

walterkozlowski commented:
@sureshkrishnan, @kumarsumangit: please see my comments I added in my review.  Once, @kumarsumangit fix the numbering and other formal things I asked there, we will be able to proceed.  Any comments from you gifs, please share at the upcoming RM meeting.

kumarsumangit commented:
@walterkozlowski  I updated the numnbered to reflect it as part of 3.5.5 . i have updated the diagram as suggested. 
I did not change any in storage except numbering of figure indices. Please review the changes.

CsatariGergely commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+<p align="center"> <img src="../figures/ch03-model-sfc-data-flow.png" alt="Data steering in Service Function Chain" Title="Data steering in Service Function Chain" width="45%"/></p>
+<p align ="center"><b>Figure 3-11c:</b> Data steering in Service Function Chain</p>
+
+- SFC classifier detects the traffic flow based on classification policies. For example, to enable SGi-Lan feature as SFC, UPF acts as SFC classifier.  UPF receives the classification policies from PCF (via SMF) as traffic steering policies. 
```
Can you please explain the acronyms at their first apperance?

kumarsumangit replied:
sure , i will bring full form. Accepted.

walterkozlowski reviewed (commented):
@kumarsumangit, thanks for the updates.  So no changes to the following sections, I take it.  Please respond to comments and suggestions from @CsatariGergely, and we will take it from there.  @sureshkrishnan, please add your comments as well. Thanks, everyone!
In general, I think we are going to have a very decent SFC sections.

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+
+The flow of steps to enable the SFC creation can be as follows:-
+
+a.	SFC orchestrator creates the SFs with help of VNF MANO or CNF MANO.
```
@CsatariGergely i illustrated the fllow common  , CNF mano means devops related with CNF and you are right kubernetes does some of functionality.  i can rename picture as cnf devops rather than MANO . 
SFC orchestrator as such not specific open source i am aware , but ONAP dublin release have once module to take care of same - https://wiki.onap.org/pages/viewpage.action?pageId=45310767 
OpenDaylight supports SFC functionality needed for sfc relaization. Openstack neutron supports the same.
Hope i am able to answer you or you are looking something specific?

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+Figure 3-11a shows a simple architecture of an SFC with multiple CNFs, as SF data plane components, along with SFC management and CNF MANO components. 
 <p align="center"> <img src="../figures/ch03-model-sfc-architecture-cnf-2.png" alt="SFC Architecture for CNF based SFs" Title="SFC Architecture for CNF based SFs" width="45%"/></p>
-<p align ="center"><b>Figure 3-11:</b> SFC Architecture for CNF based SFs</p>
+<p align ="center"><b>Figure 3-11a:</b> SFC Architecture for CNF based SFs</p>
```
@CsatariGergely i made it a b to remove collision as suggested . I think i can take up once we dont have any comments before final merge to align the figure numbering. 
For figure, i used draw.io and i have xml file to share .. but i can not have link in document and it will be lost in folder. Please let me know if you want file , i can attach in the comments section.

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+-	The service functions that are included in the SFC.
+- The routing order in the service function, if the SFC is composed of more than one service function.
+
+Figure 3-11b shows SFC creation call flow, separated logically in two steps.
```
@CsatariGergely i am not clear on what clarity is needed in figure . if you want, drawio xml file to edit , i can share.

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+-	The service functions that are included in the SFC.
+- The routing order in the service function, if the SFC is composed of more than one service function.
+
+Figure 3-11b shows SFC creation call flow, separated logically in two steps.
```
@CsatariGergely no open source for sfc classifier. in case of SGI lan feature, 5G UPF acts as classifier, which is a kind of workload and out of scope for our forum.

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+
+a.	SFC orchestrator creates the SFs with help of VNF MANO or CNF MANO.
+
+b.	SFC Rendere attaches the SFC aware interfaces at SFs to enable Service plane 
```
accepted.

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+
+b.	SFC Rendere attaches the SFC aware interfaces at SFs to enable Service plane 
+
+c.	NFVO boots up the relevant SF configurations at SF.
```
@CsatariGergely i means brings up sf configuration( workload configuration ) . It is like work load configuration to work . 
I mean here to configure the SF functions , as in case of VNF, configuration might be done by EMS / VNFM.

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+
+<a name="3.5.5.3.2"></a>
+##### 3.5.5.3.2 Updation of Service Function Chain
+SFP or SFC can be updated for various reasons and some of them are :-
```
Accepted.

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+<a name="3.5.5.3.2"></a>
+##### 3.5.5.3.2 Updation of Service Function Chain
+SFP or SFC can be updated for various reasons and some of them are :-
+- SFC controller monitors the SFP status and alerts SFC controller in case of not meeting SLA or some anamoly.
```
accepted

kumarsumangit commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -508,12 +509,74 @@ Figure 3-10 shows a simple architecture of an SFC with multiple VNFs, as SF data
+<a name="3.5.5.3.3"></a>
+##### 3.5.5.3.3 Data Steering in Service Function Chain
+
+Figure 3-11c shows traffic steering along SFP.
```
Please see above comments for sharing drawio file.

sureshkrishnan commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -298,6 +298,16 @@ It is expected that reference architectures will provide a service chain workflo
 
+<a name="3.5.1.1"></a>
+#### 3.5.1.1 Introduction 
+Service function chaining (SFC) can be visualized as layered structure where Service function plane (SFC data plane ,consists of Service function forwarder, Classifier, Service function, Service function proxy) resides over Service function overlay network. At a very high level, a service function plane is a directed acyclic graph with the composing Servie functions being the vertices.  
```
Typo Servie->Service

kumarsumangit replied:
@sureshkrishnan Hi, i see you have commented on old commit. Can you have look for recent changes as it has been modified from the commit you reviewed as content has been refined . Thanks .

sureshkrishnan commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -298,6 +298,16 @@ It is expected that reference architectures will provide a service chain workflo
+Service function chaining (SFC) can be visualized as layered structure where Service function plane (SFC data plane ,consists of Service function forwarder, Classifier, Service function, Service function proxy) resides over Service function overlay network. At a very high level, a service function plane is a directed acyclic graph with the composing Servie functions being the vertices.  
+Service function chaining utilizes a service-specific overlay that creates the service topology.  The service overlay provides service function connectivity, built "on top" of the existing network topology. It leverages various overlay network technologies (e.g., Virtual eXtensible Local Area Network (VXLAN)) for interconnecting SFC data-plane elements and allows establishing Service Function Pat ...
+
+In Overlay network, packets are routed based on networking principles as use a suitable path for the packet from source to its destination in network. 
```
Not sure what this means

"packets are routed based on networking principles as use a suitable path for the packet from source to its destination in network. "

Can you reword?

kumarsumangit replied:
@sureshkrishnan Hi, i see you have commented on old commit. Can you have look for recent changes as it has been modified from the commit you reviewed as content has been refined . Thanks .

sureshkrishnan commented on doc/ref_model/chapters/chapter03.md:
```diff
@@ -307,6 +307,36 @@ In a typical overlay network, packets are routed based on networking principles
 
+<a name="3.5.1.2"></a>
+#### 3.5.1.2 SFC Architecture
+ The SFC Architecture is built out of architectural building blocks that are logical components to relaize the service function chaining. These consist of management, control and data components. 
```
Typo relaize->realize.

kumarsumangit replied:
@sureshkrishnan Hi, i see you have commented on old commit. Can you have look for recent changes as it has been modified from the commit you reviewed as content has been refined . Thanks .

walterkozlowski commented:
@sureshkrishnan, can you please confirm that changes you proposed have been addressed.  Thanks!

pgoyal01 commented:
@CsatariGergely @sureshkrishnan Can you check if your issues have been addressed? If yes, can you resolve the conversations and requested changes? Thanks
